@AnnaShaleva dbft unit test failed 
`# github.com/ethereum/go-ethereum/eth/protocols/dbft [github.com/ethereum/go-ethereum/eth/protocols/dbft.test]
eth/protocols/dbft/handler_test.go:32:33: cannot use func(u uint64, address common.Address) bool {…} (value of type func(u uint64, address "github.com/ethereum/go-ethereum/common".Address) bool) as func(uint64, "github.com/ethereum/go-ethereum/common".Address) error value in argument to New
eth/protocols/dbft/handler_test.go:33:33: cannot use func(u uint64, address common.Address) bool {…} (value of type func(u uint64, address "github.com/ethereum/go-ethereum/common".Address) bool) as func(uint64, "github.com/ethereum/go-ethereum/common".Address) error value in argument to New
eth/protocols/dbft/handler_test.go:34:33: cannot use func(u uint64, address common.Address) bool {…} (value of type func(u uint64, address "github.com/ethereum/go-ethereum/common".Address) bool) as func(uint64, "github.com/ethereum/go-ethereum/common".Address) error value in argument to New
eth/protocols/dbft/pool_test.go:20:15: cannot use bc (variable of type *testChain) as Ledger value in argument to NewPool: *testChain does not implement Ledger (wrong type for method IsAddressAllowed)
                have IsAddressAllowed("github.com/ethereum/go-ethereum/common".Address) bool
                want IsAddressAllowed("github.com/ethereum/go-ethereum/common".Address) error
eth/protocols/dbft/pool_test.go:50:38: cannot use bc (variable of type *testChain) as Ledger value in argument to NewPool: *testChain does not implement Ledger (wrong type for method IsAddressAllowed)
                have IsAddressAllowed("github.com/ethereum/go-ethereum/common".Address) bool
                want IsAddressAllowed("github.com/ethereum/go-ethereum/common".Address) error
eth/protocols/dbft/pool_test.go:53:15: cannot use bc (variable of type *testChain) as Ledger value in argument to NewPool: *testChain does not implement Ledger (wrong type for method IsAddressAllowed)
                have IsAddressAllowed("github.com/ethereum/go-ethereum/common".Address) bool
                want IsAddressAllowed("github.com/ethereum/go-ethereum/common".Address) error
eth/protocols/dbft/pool_test.go:77:15: cannot use bc (variable of type *testChain) as Ledger value in argument to NewPool: *testChain does not implement Ledger (wrong type for method IsAddressAllowed)
                have IsAddressAllowed("github.com/ethereum/go-ethereum/common".Address) bool
                want IsAddressAllowed("github.com/ethereum/go-ethereum/common".Address) error
FAIL    github.com/ethereum/go-ethereum/eth/protocols/dbft [build failed]`

--- a/eth/protocols/dbft/handler_test.go
+++ b/eth/protocols/dbft/handler_test.go
@@ -29,9 +29,9 @@ func TestHandling(t *testing.T) {
 		key, _         = crypto.GenerateKey()
 		addr           = crypto.PubkeyToAddress(key.PublicKey)
 		bc             = &testBC{height: 10}
-		s1             = New(bc, nil, func(u uint64, address common.Address) bool { return true })
-		s2             = New(bc, nil, func(u uint64, address common.Address) bool { return true })
-		s3             = New(bc, nil, func(u uint64, address common.Address) bool { return true })
+		s1             = New(bc, nil, func(u uint64, address common.Address) error { return nil })
+		s2             = New(bc, nil, func(u uint64, address common.Address) error { return nil })
+		s3             = New(bc, nil, func(u uint64, address common.Address) error { return nil })
 		p1             = p2p.NewPeer(enode.ID{1}, "peer1", nil)
 		p2             = p2p.NewPeer(enode.ID{2}, "peer2", nil)
 		p3             = p2p.NewPeer(enode.ID{3}, "peer3", nil)

--- a/eth/protocols/dbft/pool.go
+++ b/eth/protocols/dbft/pool.go
@@ -6,6 +6,7 @@ package dbft
 import (
 	"container/list"
 	"errors"
+	"fmt"
 	"sync"
 
 	"github.com/ethereum/go-ethereum/common"
@@ -96,7 +97,7 @@ func (p *Pool) verify(m *Message) (bool, error) {
 		if errors.Is(err, ErrSyncing) {
 			return false, nil
 		}
-		return false, err
+		return false, fmt.Errorf("%w: %w", errDisallowedSender, err)
 	}
 	return true, nil
 }

--- a/eth/protocols/dbft/pool_test.go
+++ b/eth/protocols/dbft/pool_test.go
@@ -22,6 +22,12 @@ func TestAddGet(t *testing.T) {
 		m := someMessage(t, 100, bc.goodAddrs[0], bc.badKey)
 		p.testAdd(t, false, invalidSig, m)
 	})
+	t.Run("syncing", func(t *testing.T) {
+		bc.syncing = true
+		m := bc.badMessage(t, 100)
+		p.testAdd(t, false, nil, m)
+		bc.syncing = false
+	})
 	t.Run("disallowed sender", func(t *testing.T) {
 		m := bc.badMessage(t, 100)
 		p.testAdd(t, false, errDisallowedSender, m)
@@ -106,6 +112,7 @@ type testChain struct {
 	badKey    *ecdsa.PrivateKey
 	goodAddrs []common.Address
 	badAddr   common.Address
+	syncing   bool
 }
 
 var errVerification = errors.New("verification failed")
@@ -120,13 +127,16 @@ func newTestChain() *testChain {
 		badAddr:   crypto.PubkeyToAddress(bk.PublicKey),
 	}
 }
-func (c *testChain) IsAddressAllowed(u common.Address) bool {
+func (c *testChain) IsAddressAllowed(u common.Address) error {
+	if c.syncing {
+		return ErrSyncing
+	}
 	for i := range c.goodAddrs {
 		if u == c.goodAddrs[i] {
-			return true
+			return nil
 		}
 	}
-	return false
+	return errors.New("address not allowed")
 }
 func (c *testChain) BlockHeight() uint64 { return c.height }
 

